File: frontend/src/features/recovery/api/resolve_recovery.ts
import { API_CONTRACT_VERSION, type RunCommandResponse, type RunSnapshot } from "../../../api/contract";
import { requestJson } from "../../../api/client";

type RecoveryProjection = NonNullable<RunSnapshot["recovery"]>;

export function resolveRecovery(payload: { run_id: string; command_id: string; expected_version: number; target: RecoveryProjection["target"]; resolution_kind: RecoveryProjection["allowed_resolution_kinds"][number] }): Promise<RunCommandResponse> {
  return requestJson(`/api/v1/runs/${encodeURIComponent(payload.run_id)}/resolve-recovery`, {
    method: "POST",
    body: { command_id: payload.command_id, expected_version: payload.expected_version, target: payload.target, resolution_kind: payload.resolution_kind, api_contract_version: API_CONTRACT_VERSION },
  });
}

File: src/google_work_agent/api/schemas/runs/resolve_recovery.py
"""Resolve-recovery wire request."""

from google_work_agent.api.schemas.model import ContractVersionedRequest
from google_work_agent.api.schemas.runs.recovery import (
    RecoveryResolutionKindV1,
    RecoveryTargetV1,
)


class ResolveRecoveryRequestV1(ContractVersionedRequest):
    command_id: str
    expected_version: int
    target: RecoveryTargetV1
    resolution_kind: RecoveryResolutionKindV1
